subprojects/core/src/main/java/org/gradle/internal/fingerprint/impl/DefaultFingerprinterSpec.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.fingerprint.impl;
+
+import org.gradle.api.tasks.FileNormalizer;
+import org.gradle.internal.fingerprint.DirectorySensitiveNormalizer;
+import org.gradle.internal.fingerprint.DirectorySensitivity;
+import org.gradle.internal.fingerprint.FingerprinterSpec;
+
+import java.util.Objects;
+
+public class DefaultFingerprinterSpec implements FingerprinterSpec {
+    private final Class<? extends FileNormalizer> normalizer;
+    private final DirectorySensitivity directorySensitivity;
+
+    private DefaultFingerprinterSpec(Class<? extends FileNormalizer> normalizer, DirectorySensitivity directorySensitivity) {
+        this.normalizer = normalizer;
+        this.directorySensitivity = directorySensitivity;
+    }
+
+    @Override
+    public Class<? extends FileNormalizer> getNormalizer() {
+        return normalizer;
+    }
+
+    @Override
+    public DirectorySensitivity getDirectorySensitivity() {
+        return directorySensitivity;
+    }
+
+    public static FingerprinterSpec from(Class<? extends FileNormalizer> normalizer, DirectorySensitivity directorySensitivity) {
+        if (DirectorySensitiveNormalizer.class.isAssignableFrom(normalizer)) {
+            return new DefaultFingerprinterSpec(normalizer, directorySensitivity);
+        } else {
+            return new DefaultFingerprinterSpec(normalizer, DirectorySensitivity.DEFAULT);
+        }
+    }
```

Comment:
I think if we wanted to check something here, we should throw errors if the parameters are incorrect.

Reply:
At the moment, the parameters can't be incorrect.  For `DirectorySensitiveNormalizer`s the value is either `DEFAULT` (fingerprint directories) or `IGNORE_DIRECTORIES`, both of which are valid and represent different instances in the registry.  For other normalizers, the value is either `DEFAULT` (ignore directories) or `IGNORE_DIRECTORIES`, which are the same intention.  All we're doing here is making sure that for these other normalizers, we always look for the right thing in the registry (i.e. a normalizer with `DEFAULT` directory sensitivity.  See my comments here: https://github.com/gradle/gradle/pull/15147#discussion_r526416209